core/src/main/java/hudson/Functions.java

```diff
@@ -493,6 +493,16 @@ public static String getSystemProperty(String key) {
         return SystemProperties.getString(key);
     }
 
+    /**
+     * Gets the boolean system property indicated by the specific key.
+     * 
+     * Delegates to {@link SystemProperties#getBoolean(java.lang.String)}
+     */
+    @Restricted(DoNotUse.class)
+    public static boolean isSystemPropertyEnabled(String key) {
+        return SystemProperties.getBoolean(key);
```

Comment:
Oleg in https://github.com/jenkinsci/jenkins/pull/4463#pullrequestreview-357490289:

> Also keep in mind that `SystemProperties` API is rather designed for system properties which values are calculated once on restart. It involves calls like `ServletContext` checks, and I would not guarantee that it returns results immediately in all cases. Be careful when using the live switch in the frontend. If it is really needed, maybe using a `private static boolean` cache is preferable

Reply:
This was done to address the concern that it was a boolean property. We understand this approach is not optimal for doing live switches. Of course, we will choose a technique that addresses these concerns once we start working on a UI-based toggle.

Reply:
Resolving this one, feel free to reopen it if you think it deserves further discussion

Reply:
@fqueiruga My point is that the toggle should be implemented via

```
@Restricted(NoExternalUse.class)
public static /* non-final */ boolean UI_REFRESH = SystemProperties.getBoolean("jenkins.ui.refresh");
// and optionally a getter for it if accessing code can be made nicer
```

This way, you're not evaluating the potentially expensive operation on every page render but only once on startup, while retaining the ability to edit it at runtime (by setting the static var).

Reply:
I understand now. If it's not a blocker I'd rather follow up on this on a future PR rather than change this again, as it's a risky one. Feel free to reopen this convo again if you think it's necessary.

Reply:
@fqueiruga Addressed in https://github.com/jenkinsci/jenkins/pull/4463/commits/205fa3a7590439f63e8059fcee086614386f7b2e

(For which I got Felix's approval out of band)